management/backup.py

```diff
@@ -473,10 +473,91 @@ def list_target_files(config):
 			raise ValueError(e.reason)
 
 		return [(key.name[len(path):], key.size) for key in bucket.list(prefix=path)]
-
+	elif target.scheme == "b2":
+		api = b2_perform_authentication(target.username, target.password)
+		if api == None:
+			raise ValueError("B2 authentication failed for unknown reason.")
+		(auth_token, api_url) = api
+
+		bucket_id = b2_get_bucket_id(target.username, api_url, auth_token, target.hostname)
+		return b2_list_files(target.username, api_url, auth_token, bucket_id, target.path)
 	else:
 		raise ValueError(config["target"])
 
+def b2_perform_authentication(acc_id, app_key):
+	# Perform Authentication with B2 Storage. Returns a tuple
+	# containing the auth token and the API endpoint to use.
+	from urllib.request import Request, urlopen
+	from urllib.error import HTTPError
+	import base64
+	import json
+
+	# Formulate the authtoken according to B2 Storage API Spec
+	auth_token = "Basic" + base64.b64encode((acc_id + ":" + app_key).encode('utf-8')).decode('utf-8')
+	headers = {'Authorization': auth_token}
+
+	req = Request("https://api.backblazeb2.com/b2api/v1/b2_authorize_account", headers=headers)
+	try:
+		with urlopen(req) as res:
+			data = json.loads(res.read().decode('utf-8'))
+			return (data['authorizationToken'], data['apiUrl'])
+	except HTTPError as e:
+		e_data = json.loads(e.read().decode('utf-8'))
+		print(e_data)
+		if e_data['code'] == 'unauthorized':
+			raise ValueError("Invalid key pair, or B2 has not been enabled, or the user is suspended.")
```

Comment:
According to the docs, unauthorized can have different messages. It's possible they added more since this was written.
Best to put e_data['message'] in the exception message?
 
```
401 | unauthorized | B2 has not been enabled for this account
401 | unauthorized | User is in B2 suspend
401 | unauthorized | Cannot authorize domain site license account
401 | unauthorized | Invalid authorization
```